app/code/Magento/Newsletter/Model/ResourceModel/Subscriber.php

```diff
@@ -157,6 +181,52 @@ public function loadByCustomerData(\Magento\Customer\Api\Data\CustomerInterface
         return [];
     }
 
+    /**
+     * Load subscriber by customer
+     *
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @return array
+     */
+    public function loadByCustomerDataAndStore(\Magento\Customer\Api\Data\CustomerInterface $customer)
+    {
+        $select = $this->connection
+            ->select()->from($this->getMainTable())
+            ->where('customer_id=:customer_id and store_id=:store_id');
+
+        $result = $this->connection
+            ->fetchRow(
+                $select,
+                [
+                    'customer_id' => $customer->getId(),
+                    'store_id' => $customer->getStoreId()
+                ]
+            );
+
+        if ($result) {
+            return $result;
+        }
+
+        $select = $this->connection
+            ->select()
+            ->from($this->getMainTable())
+            ->where('subscriber_email=:subscriber_email and store_id=:store_id');
+
+        $result = $this->connection
+            ->fetchRow(
+                $select,
+                [
+                    'subscriber_email' => $customer->getEmail(),
+                    'store_Id' => $customer->getStoreId()
```

Comment:
Typo. Should be 'store_id' instead of 'store_Id'.
Leads to exception.

Reply:
Thanks for the heads_up. I am away but I will try to fix this and resubmit.